http-server/src/main/java/io/micronaut/http/server/exceptions/response/DefaultJsonBodyErrorResponseProvider.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.server.exceptions.response;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.hateoas.JsonError;
+import io.micronaut.http.hateoas.Link;
+import io.micronaut.http.hateoas.Resource;
+import io.micronaut.json.JsonConfiguration;
+import jakarta.inject.Singleton;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Default implementation of {@link JsonBodyErrorResponseProvider} which returns a {@link JsonError}.
+ *
+ * @since 4.7.0
+ */
+@Internal
+@Singleton
+class DefaultJsonBodyErrorResponseProvider implements JsonBodyErrorResponseProvider<JsonError> {
```

Comment:
should be `final` if possible

Reply:
https://github.com/micronaut-projects/micronaut-core/pull/11210/commits/4ded5ae25b2f9e0aba7744bd7505211e5ba6883e